### PR TITLE
Patch the nearest positions equidistance detection

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #23585 

need this for Javi work asap

### CODE REVIEW

**Review by lindsayad — CHANGES_REQUESTED** [review]
How does this work? Code comment would be appreciated

**Comment by GiudGiud** [comment]
added some

**Comment by lindsayad** [comment]
I still don't understand why this now avoids false positives

**Comment by GiudGiud** [comment]
Before we were trying to detect them live in the for loop
so if you look at positions P3 and P4 and they are equidistant to the target (and smaller distance than P1 and P2), we would send a warning. 

but if P5 is actually the closest, then you sent a warning for nothing. P3 and P4 are not the closest positions, who cares that they are equidistant.

now we wait until we have looked at all positions. then send a warning if the closest was also reported as a conflict

File: framework/src/positions/Positions.C [hunk 1]
```diff
@@ -92,10 +94,18 @@ Positions::getNearestPosition(const Point & target, const bool initial) const
...
-              " and " + Moose::stringify(_positions[nearest_index]),
-          " at a distance of " + std::to_string(std::sqrt(nearest_distance_sq)));
+      conflict_index = std::make_pair(cast_int<int>(i), cast_int<int>(nearest_index));
+  }
+
+  // If the closest positions was a conflict, report a warning
+  if (cast_int<int>(nearest_index) == conflict_index.first ||
+      cast_int<int>(nearest_index) == conflict_index.second)
+  {
+    mooseWarning("Search for nearest position found several matches: " +
```
Comment (lindsayad) [inline framework/src/positions/Positions.C:104]:
```suggestion
    mooseWarning("Search for nearest position found at least two matches: " +
```

File: framework/src/positions/Positions.C [hunk 2]
```diff
@@ -92,10 +94,18 @@ Positions::getNearestPosition(const Point & target, const bool initial) const
...
-      mooseWarning(
-          "Search for nearest position found several matches: " + Moose::stringify(_positions[i]) +
-              " and " + Moose::stringify(_positions[nearest_index]),
-          " at a distance of " + std::to_string(std::sqrt(nearest_distance_sq)));
+      conflict_index = std::make_pair(cast_int<int>(i), cast_int<int>(nearest_index));
+  }
+
+  // If the closest positions was a conflict, report a warning
+  if (cast_int<int>(nearest_index) == conflict_index.first ||
+      cast_int<int>(nearest_index) == conflict_index.second)
```
Comment (lindsayad) [inline framework/src/positions/Positions.C:102]:
```suggestion
  if (cast_int<int>(nearest_index) == conflict_index.second)
```
It should only be possible to match the second right?

File: framework/src/positions/Positions.C [hunk 3]
```diff
@@ -92,10 +94,18 @@ Positions::getNearestPosition(const Point & target, const bool initial) const
...
     else if (MooseUtils::absoluteFuzzyEqual((_positions[i] - target).norm_sq(),
                                             nearest_distance_sq))
-      mooseWarning(
-          "Search for nearest position found several matches: " + Moose::stringify(_positions[i]) +
-              " and " + Moose::stringify(_positions[nearest_index]),
-          " at a distance of " + std::to_string(std::sqrt(nearest_distance_sq)));
+      conflict_index = std::make_pair(cast_int<int>(i), cast_int<int>(nearest_index));
+  }
+
+  // If the closest positions was a conflict, report a warning
```
Comment (lindsayad) [inline framework/src/positions/Positions.C:100]:
```suggestion
  // If there were multiple "closest" positions, report a warning
```

File: framework/src/positions/Positions.C (hunk 2 above)
Comment (GiudGiud) [reply framework/src/positions/Positions.C:102]: sure

**Review by lindsayad — DISMISSED** [review]

**Comment by GiudGiud** [comment]
looks like not

**Comment by lindsayad** [comment]
That wasn't the problem. It was the change in error message

**Comment by lindsayad** [comment]
From https://civet.inl.gov/job/1685443/:
```
Search for nearest position found at least two matches
```

**Comment by lindsayad** [comment]
Current regex is
```
Search for nearest position found several matches
```

**Comment by lindsayad** [comment]
You could revert that change if you want, but I like matching against only what we think we should be matching (in fact I myself might add an assertion to make sure what I think is happening is happening)

**Comment by GiudGiud** [comment]
regex changed

**Review by lindsayad — APPROVED** [review]

**Pull request merged**